controlplane/pkg/nsmd/registry.go

```diff
@@ -75,7 +80,8 @@ func (es *registryServer) RegisterNSE(ctx context.Context, request *registry.NSE
 	span.LogObject("registration", reg)
 	return reg, nil
 }
-func (es *registryServer) RegisterNSEWithClient(ctx context.Context, request *registry.NSERegistration, client registry.NetworkServiceRegistryClient) (*registry.NSERegistration, error) {
+
+func (es *registryServer) RegisterNSEWithClient(ctx context.Context, request *registry.NSERegistration, client registry.NetworkServiceRegistryClient, workspace *Workspace) (*registry.NSERegistration, error) {
```

Comment:
I don't quite like the idea, because up to the point of call of RegisterNSEWithClient the workspace  is passed as an explicit argument, changing that seems unnecessary.